src/main/java/org/aludratest/service/flatfile/impl/DefaultFlatFileAction.java

```diff
@@ -192,6 +192,11 @@ public String toString() {
         }
 
         @Override
+        public boolean equals(Object other) {
+            return internalKey.equals(other);
```

Comment:
Wrong implementation. other is a WriterKey, never an "internal key". Check for null, for other being == this, then check other.getClass() == getClass(), and **then** compare other.internalKey to internalKey.